iib/workers/tasks/build.py

```diff
@@ -699,16 +699,39 @@ def get_index_image_info(overwrite_from_index_token, from_index=None, default_oc
     return result
 
 
+def get_binary_image_from_config(ocp_version, distribution_scope, binary_image_config={}):
+    """
+    Determine the binary image to be used to build the index image.
+
+    :param str ocp_version: the ocp_version label value of the index image.
+    :param str distribution_scope: the distribution_scope label value of the index image.
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    :return: pull specification of the binary_image to be used for this build.
+    :rtype: str
+    :raises IIBError: when the config value for the ocp_version and distribution_scope is missing.
+    """
+    binary_image = binary_image_config.get(distribution_scope, {}).get(ocp_version, None)
+    if not binary_image:
+        raise IIBError(
+            'No binary image found in IIB_BINARY_IMAGE_CONFIG for'
```

Comment:
Can you make this more user friendly? The user doesn't care about `IIB_BINARY_IMAGE_CONFIG` being a thing. They just need to know IIB couldn't automatically determine the appropriate binary image to use.

Reply:
Done